tests/returntypes/test_tensorflow_qnode_new.py

```diff
@@ -0,0 +1,1569 @@
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        from pennylane import numpy as anp
+
+        a = anp.array(0.1, requires_grad=True)
+
+        res1 = circuit(a)
+        grad_fn = qml.grad(circuit)
+        grad1 = grad_fn(a)
+
+        # switch to TF interface
+        circuit.interface = "tf"
+
+        a = tf.Variable(0.1, dtype=tf.float64)
+
+        with tf.GradientTape() as tape:
+            res2 = circuit(a)
+
+        grad2 = tape.gradient(res2, a)
+        assert np.allclose(res1, res2, atol=tol, rtol=0)
+        assert np.allclose(grad1, grad2, atol=tol, rtol=0)
+
+    def test_drawing(self, dev_name, diff_method, mode):
+        """Test circuit drawing when using the TF interface"""
+
+        x = tf.Variable(0.1, dtype=tf.float64)
+        y = tf.Variable([0.2, 0.3], dtype=tf.float64)
+        z = tf.Variable(0.4, dtype=tf.float64)
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, interface="tf", diff_method=diff_method, mode=mode)
+        def circuit(p1, p2=y, **kwargs):
+            qml.RX(p1, wires=0)
+            qml.RY(p2[0] * p2[1], wires=1)
+            qml.RX(kwargs["p3"], wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.state()
+
+        result = qml.draw(circuit)(p1=x, p3=z)
+        expected = "0: ──RX(0.10)──RX(0.40)─╭●─┤  State\n" "1: ──RY(0.06)───────────╰X─┤  State"
+        assert result == expected
+
+    def test_jacobian(self, dev_name, diff_method, mode, mocker, tol):
+        """Test jacobian calculation"""
+        if diff_method == "parameter-shift":
+            spy = mocker.spy(qml.gradients.param_shift, "transform_fn")
+        elif diff_method == "finite-diff":
+            spy = mocker.spy(qml.gradients.finite_diff, "transform_fn")
+
+        a = tf.Variable(0.1, dtype=tf.float64)
+        b = tf.Variable(0.2, dtype=tf.float64)
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, diff_method=diff_method, mode=mode, interface="tf")
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))
+
+        with tf.GradientTape() as tape:
+            res = circuit(a, b)
+            res = tf.stack(res)
+
+        assert circuit.qtape.trainable_params == [0, 1]
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2,)
+
+        expected = [tf.cos(a), -tf.cos(a) * tf.sin(b)]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = tape.jacobian(res, [a, b])
+        expected = [[-tf.sin(a), tf.sin(a) * tf.sin(b)], [0, -tf.cos(a) * tf.cos(b)]]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        if diff_method in ("parameter-shift", "finite-diff"):
+            spy.assert_called()
+
+    @pytest.mark.xfail

```

Comment:
I don't know what this test is for (I just copied it from the original file). It seems it relies on `qnode.to_tf` which doesn't exist anymore. Do you know anything about it?

Reply:
`qnode.to_tf` used to transform the QNode to a TF QNode - equivalent to applying the interface. I think that line and the following where `circuit` is inspected for its dtype can be removed. Otherwise the test case seems to have merit, though I'm not sure if we have been consistent with preserving datatypes.

Reply:
Added this test back.